internal/provider/resource_time_offset.go

```diff
@@ -0,0 +1,455 @@
+				Description: "Number of seconds to offset the base timestamp. At least one of the 'offset_' arguments must be configured.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				//AtLeastOneOf: []string{
+				//	"offset_days",
+				//	"offset_hours",
+				//	"offset_minutes",
+				//	"offset_months",
+				//	"offset_seconds",
+				//	"offset_years",
+				//},
+			},
+			"offset_years": {
+				Description: "Number of years to offset the base timestamp. At least one of the 'offset_' arguments must be configured.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				//AtLeastOneOf: []string{
+				//	"offset_days",
+				//	"offset_hours",
+				//	"offset_minutes",
+				//	"offset_months",
+				//	"offset_seconds",
+				//	"offset_years",
+				//},
+			},
+			"rfc3339": {
+				Description: "RFC3339 format of the offset timestamp, e.g. `2020-02-12T06:36:13Z`.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"second": {
+				Description: "Number second of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"unix": {
+				Description: "Number of seconds since epoch time, e.g. `1581489373`.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"year": {
+				Description: "Number year of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"id": {
+				Description: "RFC3339 format of the offset timestamp, e.g. `2020-02-12T06:36:13Z`.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+		},
+	}, nil
+}
+
+func (t timeOffsetResourceType) NewResource(ctx context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return &timeOffsetResource{}, nil
+}
+
+var (
+	_ tfsdk.Resource                = (*timeOffsetResource)(nil)
+	_ tfsdk.ResourceWithImportState = (*timeOffsetResource)(nil)
+)
+
+type timeOffsetResource struct {
+}
+
+func (t timeOffsetResource) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+	id := req.ID
+
+	idParts := strings.Split(id, ",")
+
+	if len(idParts) != 7 {
+		resp.Diagnostics.AddError(
+			"Unexpected Format of ID",
+			fmt.Sprintf("Unexpected format of ID (%q), expected BASETIMESTAMP,YEARS,MONTHS,DAYS,HOURS,MINUTES,SECONDS", id))
+
+		return
+	}
+
+	if idParts[0] == "" || (idParts[1] == "" && idParts[2] == "" && idParts[3] == "" && idParts[4] == "" && idParts[5] == "" && idParts[6] == "") {
+		resp.Diagnostics.AddError(
+			"Unexpected Format of ID",
+			fmt.Sprintf("Unexpected format of ID (%q), expected BASETIMESTAMP,YEARS,MONTHS,DAYS,HOURS,MINUTES,SECONDS where at least one offset value is non-empty", id))
+
+		return
+	}
+
+	baseRfc3339 := idParts[0]
+
+	offsetYears := types.Int64{Null: true}
+	if idParts[1] != "" {
+		offset, err := strconv.ParseInt(idParts[1], 10, 64)
+		if err != nil {
+			// return diagnostic
+		}
+
+		offsetYears.Value = offset
+		offsetYears.Null = false
+	}
```

Comment:
We can bundle this up into a `func` that we can re-use, perhaps something like:
```go
func offsetToInt64(offsetStr string) (types.Int64, error) {
	offset := types.Int64{Null: true}
	
	if offsetStr != "" {
		offsetInt, err := strconv.ParseInt(offsetStr, 10, 64)
		if err != nil {
			return offset, fmt.Errorf("could not parse offset (%q) as int: %w", offsetStr, err)
		}

		offset.Value = offsetInt
		offset.Null = false
	}
	
	return offset, nil
}
```

If we receive an error, we can generate a diagnostic indicating which offset was problematic and we can also unwrap the error (`unwrappedErr := errors.Unwrap(err)`) in case we need it for logging etc.